src/expr/src/scalar/func/impls/timestamp.rs

```diff
@@ -81,6 +84,7 @@ sqlfunc!(
     #[sqlname = "timestamp_with_time_zone_to_timestamp"]
     #[preserves_uniqueness = true]
     #[inverse = to_unary!(super::CastTimestampToTimestampTz)]
+    #[is_monotone = true]
     fn cast_timestamp_tz_to_timestamp(
```

Comment:
I think this one is correct! Some relevant definitions:

```rust
    /// Returns a view to the naive UTC datetime.
    #[inline]
    pub fn naive_utc(&self) -> NaiveDateTime {
        self.datetime
    }

    /// Returns a view to the naive local datetime.
    #[inline]
    pub fn naive_local(&self) -> NaiveDateTime {
        self.datetime + self.offset.fix()
    }
```

ie. the stored datetime is already in UTC; it's stored alongside a timezone which is applied whenever you ask for the local time. And ordering just compares the UTC datetimes, ignoring the timezone.

Reply:
Oh, ok! I was confused because the `DateTime` struct itself has an `offset` field, so I thought the offset might have different values. (In that case, the offset could flip the comparison, and then it could happen that 
`a.datetime + a.offset < b.datetime + b.offset`, but
`cast_that_forgets_the_offset(a) > cast_that_forgets_the_offset(b)`) 

But reading the [relevant Materialize docs](https://materialize.com/docs/sql/types/timestamp/), it seems the offset is always 0, if I understand correctly? You might want to add a comment there to explain this. Also, I'm not sure whether this behavior might change in the future, i.e., the `offset` might become used, and then someone will need to not forget to adjust `is_monotone`. 

@mjibson, what do you think?
- Is the `offset` field really always 0 here?
- Will it always stay this way? If not, where should we add a comment to avoid the `is_monotone` annotation suddenly becoming invalid in the future? Or if the `offset` field would become used, then this cast function would also start including the `offset` field (e.g., call `naive_local` instead of `naive_utc`), and then the cast would stay monotone?

Reply:
> nor comparison even look the offset field at all

Ah, I get it now, this was the critical part that I missed. Sorry for being slow!

Reply:
Not at all... this stuff is pretty confusing!